tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp

```diff
@@ -861,17 +861,17 @@ std::string SwiftLangSupport::resolvePathSymlinks(StringRef FilePath) {
   if (fileHandle == INVALID_HANDLE_VALUE)
     return InputPath;
 
-  bool success = GetFinalPathNameByHandleW(fileHandle, full_path, MAX_PATH,
+  DWORD numChars = GetFinalPathNameByHandleW(fileHandle, full_path, MAX_PATH,
                                             FILE_NAME_NORMALIZED);
   CloseHandle(fileHandle);
   std::string utf8Path;
-  if (success) {
+  if (numChars) {
```

Comment:
This should be `numChars <= MAX_PATH`.  Upon a failure due to an inadequately sized buffer, the number of TCHARs required is returned.